.circleci/config.yml

```diff
@@ -72,181 +86,146 @@ executors:
-  # ensures that Go dependencies are up-to-date
-  dependencies:
-    executor: golang
+  # docker build image
+  docker-build:
+    executor: golang_large
+    parameters:
+      target:
+        description: "which target to build"
+        type: string
     steps:
       - go_restore_cache
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - when:
+          condition:
+            equal: [injector, << parameters.target >>]
+          steps:
+            - ebpf_builder_restore_cache
       - run:
-          name: Ensure dependencies are up-to-date
-          command: |
-            go mod tidy
-            go mod vendor
-            git diff --exit-code
-  # this will fail if changes in packages are checked into git but the updated manifests are not
-  manifests:
-    executor: golang
+          name: Build docker image for <<parameters.target>>
+          command: make docker-build-<<parameters.target>> USE_VOLUMES=true
+      - when:
+          condition:
+            equal: [injector, << parameters.target >>]
+          steps:
+            - ebpf_builder_save_cache
+      - persist_to_workspace:
+          # Must be an absolute path, or relative path from working_directory. This is a directory on the container which is
+          # taken to be the root directory of the workspace.
+          root: /home/circleci/go
+          # Must be relative path from root
+          paths:
+            - src/github.com/DataDog/chaos-controller/bin/<< parameters.target >>/<< parameters.target >>.tar.gz
+  go-make:
+    executor: <<parameters.executor>>
+    parameters:
+      target:
+        description: "which make target to call"
+        type: string
+      diff-fails:
+        description: "should git diff exit code be checked following ran command"
+        type: boolean
+        default: false
+      executor:
+        type: string
+        default: golang
     steps:
       - go_restore_cache
       - run:
-          name: Ensure manifests are checked into git
-          command: |
-            make manifests
-            git diff --exit-code
-  # vets the code
-  vet:
-    executor: golang
-    steps:
-      - go_restore_cache
-      - run: make vet
-  # formats the code to ensure it complies with Go standards
-  fmt:
-    executor: golang
-    steps:
-      - go_restore_cache
-      - run: make fmt
-  # runs the golangci-lint linters
-  lint:
-    executor: golang_large
-    steps:
-      - go_restore_cache
-      - run: make lint
+          name: run make <<parameters.target>>
+          command: make <<parameters.target>>
+      - when:
+          condition: << parameters.diff-fails >>
+          steps:
+            - run:
+                name: Check diffs following generate
+                command: git diff --exit-code
   # runs the tests
   test:
-    executor: golang
+    executor: golang_large
     steps:
       - go_restore_cache
       - run: make test
+      - store_test_results:
+          path: report.xml
```

Comment:
upload test reports to circle ci to enjoy [this](https://app.circleci.com/insights/github/DataDog/chaos-controller/workflows/test_and_build/tests?branch=matthieu.bono/CHAOS-503/upgrade-gingko-v2)